app/models/article.rb

```diff
@@ -416,6 +416,22 @@ def access_by?(user)
     user.admin? || user_id == user.id
   end
 
+  def merge_with(other_article_id)
+    if !Article.exists?(other_article_id) || other_article_id == self.id
+      return self
+    else
+      other_article = Article.find(other_article_id)
```

Comment:
On line 420 I'm making sure that the `other_article` exists and that it's not the same one as the current article. The current article should always exist since it's the article you're editing.